include/deal.II/sundials/ida_interface.h

```diff
@@ -0,0 +1,476 @@
+//-----------------------------------------------------------
+//
+//    Copyright (C) 2017 by the deal.II authors
+//
+//    This file is part of the deal.II library.
+//
+//    The deal.II library is free software; you can use it, redistribute
+//    it, and/or modify it under the terms of the GNU Lesser General
+//    Public License as published by the Free Software Foundation; either
+//    version 2.1 of the License, or (at your option) any later version.
+//    The full text of the license can be found in the file LICENSE at
+//    the top level of the deal.II distribution.
+//
+//---------------------------------------------------------------
+
+#ifndef dealii_sundials_ida_interface_h
+#define dealii_sundials_ida_interface_h
+
+#include <deal.II/base/config.h>
+#ifdef DEAL_II_WITH_SUNDIALS
+
+#include <deal.II/base/logstream.h>
+#include <deal.II/base/exceptions.h>
+#include <deal.II/base/parameter_handler.h>
+#include <deal.II/base/conditional_ostream.h>
+#include <deal.II/base/mpi.h>
+#include <deal.II/lac/vector.h>
+#include <deal.II/lac/vector_view.h>
+
+
+#include <ida/ida.h>
+#include <ida/ida_spils.h>
+#include <ida/ida_spgmr.h>
+#include <ida/ida_spbcgs.h>
+#include <ida/ida_sptfqmr.h>
+#include <nvector/nvector_serial.h>
+#include <sundials/sundials_math.h>
+#include <sundials/sundials_types.h>
+
+#include <memory>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace SUNDIALS
+{
+
+  /** Interface to SUNDIALS IDA library.
+   *
+   * The class IDAInterface is a wrapper to the Implicit
+   * Differential-Algebraic solver which is a general purpose solver for
+   * systems of Differential-Algebraic Equations (DAEs).
+   *
+   * The user has to provide the implmentation of the following std::functions:
+   *  - create_new_vector;
+   *  - residual;
+   *  - setup_jacobian;
+   *  - solve_jacobian_system;
+   *  - output_step;
+   *  - solver_should_restart;
+   *  - differential_components.
+   *
+   * Citing from the SUNDIALS documentation:
+   *
+   *   Consider a system of Differential-Algebraic Equations written in the
+   *   general form
+   *
+   * \f[
+   *   \begin{cases}
+   *       F(t,y,\dot y) = 0\, , \\
+   *       y(t_0) = y_0\, , \\
+   *       \dot y (t_0) = \dot y_0\, .
+   *   \end{cases}
+   * \f]
+   *
+   * where \f$y,\dot y\f$ are vectors in \f$\R^n\f$, \f$t\f$ is often the time (but can
```

Comment:
You don't need to use `\f` (in fact, it is wrong) since we automatically expand `$` to `\f$`.

Reply:
I copyed from deal2lkit... there we don't have this auto-magic expansion... :) I'll fix this.